juju/client/overrides.py

```diff
@@ -43,3 +48,35 @@ def __init__(self, deltas=None):
     @classmethod
     def from_json(cls, data):
         return cls(deltas=data)
+
+
+class ResourcesFacade(Type):
+    """Patch parts of ResourcesFacade to make it work.
+    """
+
+    @ReturnMapping(_client.AddPendingResourcesResult)
+    async def AddPendingResources(self, application_tag, charm_url, resources):
```

Comment:
We should start with this but open a bug to the juju team to have these put under the right api code so the generator will pick these up